record_a.go

```diff
@@ -0,0 +1,134 @@
+package ibclient
+
+import (
+	"fmt"
+	_ "time"
+)
+
+type RecordAOperations interface {
+	CreateARecord(recA RecordA) (*RecordA, error)
+	GetARecord(recA RecordA) (*[]RecordA, error)
+	DeleteARecord(recA RecordA) (string, error)
+	UpdateARecord(recA RecordA) (*RecordA, error)
+}
+
+// RecordA represents NIOS DNS A Record Object
+type RecordA struct {
+	IBBase   `json:"-"`
+	Ref               string `json:"_ref,omitempty"`
+	Ipv4Addr          string `json:"ipv4addr,omitempty"`
+	Name              string `json:"name,omitempty"`
+	View              string `json:"view,omitempty"`
+	Zone              string `json:"zone,omitempty"`
+	Ea                EA     `json:"extattrs,omitempty"`
+	NetView           string `json:"omitempty"`
+	Cidr              string `json:"omitempty"`
+	AddEA             EA     `json:"omitempty"`
+	RemoveEA          EA     `json:"omitempty"`
+	CreationTime      int    `json:"creation_time,omitempty"`
+	Comment           string `json:"comment,omitempty"`
+	Creator           string  `json:"creator,omitempty"`
+	DdnsProtected     bool    `json:"ddns_protected,omitempty"`
+	DnsName           string  `json:"dns_name,omitempty"`
+	ForbidReclamation bool  `json:"forbid_reclamation,omitempty"`
+	Reclaimable       bool     `json:"reclaimable,omitempty"`
+ 	Ttl               uint     `json:"ttl,omitempty"`
+	UseTtl            bool     `json:"use_ttl,omitempty"`
+}
+
+// NewRecordA creates a new A Record type with objectType and returnFields 
+func NewRecordA(ra RecordA) *RecordA {
+	res := ra
+	res.objectType = "record:a"
+
+	res.returnFields = []string{ "ipv4addr", "name", "view", "zone","extattrs","comment","creation_time",
+				     "creator","ddns_protected","dns_name","cloud_info","forbid_reclamation","last_queried",
+				      "reclaimable","ttl","use_ttl","aws_rte53_record_info","ddns_principal","disable","discovered_data","ms_ad_user_data"}
+	return &res
+}
+
+// CreateARecord takes Name, Ipv4Addr and View of the record to create A Record
+// Optional fields: NetView, Ea, Cidr
+// Allocates the next available IPv4Addr if IPv4Addr is not passed
+func (objMgr *ObjectManager) CreateARecord(recA RecordA) (*RecordA, error) {
+	recA.Ea = objMgr.extendEA(recA.Ea)
+	recordA := NewRecordA(recA)
+	if recA.Ipv4Addr == "" {
+		recordA.Ipv4Addr = fmt.Sprintf("func:nextavailableip:%s,%s", recA.Cidr, recA.NetView)
+	} else {
+		recordA.Ipv4Addr = recA.Ipv4Addr
+	}
+	ref, err := objMgr.connector.CreateObject(recordA)
+	recordA.Ref = ref
+	return recordA, err
+}
+
+// GetARecord by passing Name, reference ID, IP Address or DNS View
+// If no arguments are passed then, all the records are returned
+func (objMgr *ObjectManager) GetARecord(recA RecordA) (*[]RecordA, error) {
+
+	var res []RecordA
+	recordA := NewRecordA(recA)
+	var err error
+	if len(recA.Ref)>0 {
+		err = objMgr.connector.GetObject(recordA, recA.Ref, &recordA)
+		res = append(res,*recordA)
+
+	} else {
+		recordA = NewRecordA(recA)
```

Comment:
I think this is a redundant line of code @Anaghakh . Can you once verify if so.

Reply:
Yes, it was a redundant line. I've removed the line of code in both GET and DELETE. Thank you.